packages/appservice-rest/src/Arm/azure-arm-resource.ts

```diff
@@ -11,11 +11,14 @@ export class Resources {
         this._client = new ServiceClient(endpoint, 30);
     }
 
-    public async getResources(resourceType: string, resourceName: string) {
+    public async getResources(resourceType: string, resourceName: string, resourceGroupName?:string) {
+        var queryParameters = (resourceGroupName != undefined && resourceGroupName) ? `$filter=resourceType EQ \'${encodeURIComponent(resourceType)}\' AND resourceGroup EQ \'${encodeURIComponent(resourceGroupName)}\' AND name EQ \'${encodeURIComponent(resourceName)}\'` 

```

Comment:
**Suggestion**

var queryParameters = (resourceGroupName != undefined && resourceGroupName) ? `$filter=resourceType EQ \'${encodeURIComponent(resourceType)}\' AND resourceGroup EQ \'${encodeURIComponent(resourceGroupName)}\' AND name EQ \'${encodeURIComponent(resourceName)}\'` 
        :`$filter=resourceType EQ \'${encodeURIComponent(resourceType)}\' AND name EQ \'${encodeURIComponent(resourceName)}\'`;

Instead of writing this query as above, would it be better if we write as below

var queryParameters  = `$filter=resourceType EQ \'${encodeURIComponent(resourceType)}\' AND name EQ \'${encodeURIComponent(resourceName)}\'`

if (resourceGroupName != undefined && resourceGroupName) {
   queryParameters.append(`AND resourceGroup EQ \'${encodeURIComponent(resourceGroupName)}`) 
}

It will reduce one line and resourceGroupName will look like more optional.

Reply:
I thought about it too, I was just wondering if they will be some issue with concatenation. Will try and update the PR.

Reply:
Sure @shpraka, thanks..

Reply:
Updated the PR.